### PR TITLE
Password validation

### PR DESCRIPTION
**Pull request by buger**
Adds 4 new attribute checks:

* `minNumeric` - minimum number of numeric characters in string
* `minSpecial` -  minimum require number of special characters in string
* `multiCase` - require at least one lower and upper character
* `disableSequential` - disable sequential characters like `abc`, `123` or `ccc`

### CODE REVIEW

File: validation.go [hunk 1]
```diff
@@ -718,6 +718,72 @@ func (v *subSchema) validateString(currentSubSchema *subSchema, value interface{
...
+				new(StringMultiCaseError),
+				context,
+				value,
+				ErrorDetails{"multi_case": currentSubSchema.multiCase},
+			)
+		}
+	}
+
+	if currentSubSchema.disableSequential {
+		re := regexp.MustCompile("(?i)(abc|bcd|cde|def|efg|fgh|ghi|hij|ijk|jkl|klm|lmn|mno|nop|opq|pqr|qrs|rst|stu|tuv|uvw|vwx|wxy|xyz|012|123|234|345|456|567|678|789)+")
```
Comment (mvdan) [inline validation.go]: nitpicking, but do you really need the `+` here?

File: validation.go [hunk 2]
```diff
@@ -718,6 +718,72 @@ func (v *subSchema) validateString(currentSubSchema *subSchema, value interface{
...
+
+		var seq []string
+
+		for i := range stringValue {
+			if i > 2 {
```
Comment (mvdan) [inline validation.go]: invert and continue to unindent

File: validation.go [hunk 3]
```diff
@@ -718,6 +718,72 @@ func (v *subSchema) validateString(currentSubSchema *subSchema, value interface{
...
+				context,
+				value,
+				ErrorDetails{"min_special": *currentSubSchema.minSpecial},
+			)
+		}
+	}
+
+	if currentSubSchema.multiCase {
+		reL := regexp.MustCompile("[0-9a-z]")
+		reU := regexp.MustCompile("[A-Z]")
```
Comment (mvdan) [inline validation.go]: This will not work with unicode - for example, `foobÀr` would not be accepted.

**Review by mvdan — APPROVED** [review]

**Comment by buger** [comment]
Thanks for the tip, I actually never used unicode character classes. Really cool feature.

**Comment by mvdan** [comment]
Yep, they are really useful. graphemica is also very useful, you can put whatever unicode char you want in the url: http://graphemica.com/%EF%BC%9F

**Pull request merged**